### PR TITLE
Host docs on Read the Docs

### LINKED ISSUE #5266: Numba documentation page does not provide different versions
## Reporting a bug

- [ ] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [ ] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

Numba website used to have a drop down menu(When I click the version at the top of webpage) to chose different doc version. But that functionality is gone now.
Should be able to chose different versions on the website instead of typing the version I want directly on the URL.

### PR DESCRIPTION
**Pull request by AndrewEckart**
Adds configuration and requirements files to enable building and hosting the docs on Read the Docs. Results can be viewed at https://numba.readthedocs.io/en/readthedocs/index.html.

Fixes #5266. I'm not sure what additional cleanup should be done (eliminating the "bespoke doc automation system"?). I've added @esc and @stuartarchibald as maintainers on RTD so they can give admin access to anyone else who needs it.

### CODE REVIEW

**Comment by AndrewEckart** [comment]
@StanczakDominik Yeah, I can do that. One question - should I activate the release branches (e.g. `release0.48`), or use the tags (`0.48.0`, commit cc7e7c7). I don't think the builds will succeed unless these config files are merged into each version.

**Comment by stuartarchibald** [comment]
Thanks for the patch. I've queued it for review.

**Comment by stuartarchibald** [comment]
@StanczakDominik thanks for providing feedback on this.

**Comment by stuartarchibald** [comment]
@AndrewEckart thanks for doing this and adding us as maintainers!

**Comment by stuartarchibald** [comment]
> Adds configuration and requirements files to enable building and hosting the docs on Read the Docs. Results can be viewed at https://numba.readthedocs.io/en/readthedocs/index.html.

This looks good, thanks.

> Fixes #5266. I'm not sure what additional cleanup should be done (eliminating the "bespoke doc automation system"?). 

I think the "bespoke doc automation system" is a lot less troublesome now we have a more standard style to the docs? @seibert any view on this?

> I've added @esc and @stuartarchibald as maintainers on RTD so they can give admin access to anyone else who needs it.

Thanks for permitting this, we'll add the rest of the maintainers on patch approval.

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Required
+version: 2
+
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+  configuration: docs/source/conf.py
+
+# Build documentation with MkDocs
+#mkdocs:
+#  configuration: mkdocs.yml
```
Comment (stuartarchibald) [inline .readthedocs.yml:13]: Dead code?

File: .readthedocs.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# Optionally build your docs in additional formats such as PDF
+formats:
+  - pdf
+
+# Optionally set the version of Python and requirements required to build your docs
+python:
+  version: 3.7
```
Comment (stuartarchibald) [inline .readthedocs.yml:21]: Is this needed? Docs should build ok under any supported python version?

File: docs/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+numpydoc
```
Comment (stuartarchibald) [inline docs/requirements.txt:1]: Is `pygments` present from some other dependency chain? I've a recollection it's needed but guess it's coming in from somewhere else as the demo URL renders fine.

File: .readthedocs.yml (hunk 1 above)
Comment (StanczakDominik) [reply .readthedocs.yml:13]: Most certainly dead code.

File: .readthedocs.yml (hunk 2 above)
Comment (StanczakDominik) [reply .readthedocs.yml:21]: They should, but [according to docs](https://docs.readthedocs.io/en/stable/config-file/v2.html#python-version) the default is just 3, and it's probably better to be explicit about it :)

File: docs/requirements.txt (hunk 3 above)
Comment (StanczakDominik) [reply docs/requirements.txt:1]: Looks like RTD downloads Sphinx by default, and `pygments` are a dependency there. Ends up working okay.

File: .readthedocs.yml (hunk 1 above)
Comment (AndrewEckart) [reply .readthedocs.yml]: Removed.

File: .readthedocs.yml (hunk 2 above)
Comment (stuartarchibald) [reply .readthedocs.yml:17]: ah, thanks, best leave this in then!

File: docs/requirements.txt (hunk 3 above)
Comment (stuartarchibald) [reply docs/requirements.txt:1]: Then I guess leave as-is until proven otherwise :)

**Comment by seibert** [comment]
+1 from me.  Our current doc update system is a cron job pushing changes to a Github pages site, along with manual work to generate and push tagged releases.  It is easy to turn off, and as a followup to this, we'll need to update the Numba homepage to link to RTD.

**Comment by sklam** [comment]
We can now switch the rtd config to point to this repo.

**Comment by AndrewEckart** [comment]
> We can now switch the rtd config to point to this repo.

Done, the latest build succeeded and is up at https://numba.readthedocs.io/en/latest/

**Comment by stuartarchibald** [comment]
@AndrewEckart Hi, was just debugging some issue with the RTD builds and noticed you are listed in the maintainers https://readthedocs.org/projects/numba/, I guess this being a result of you kindly instigating this change. Was wondering if you are happy to now hand over the maintenance entirely to the Numba maintainers (i.e. rescind your current Numba RTD maintainer status)? Many thanks :)

**Comment by AndrewEckart** [comment]
> @AndrewEckart Hi, was just debugging some issue with the RTD builds and noticed you are listed in the maintainers https://readthedocs.org/projects/numba/, I guess this being a result of you kindly instigating this change. Was wondering if you are happy to now hand over the maintenance entirely to the Numba maintainers (i.e. rescind your current Numba RTD maintainer status)? Many thanks :)

Yes, of course!

**Comment by stuartarchibald** [comment]
> > @AndrewEckart Hi, was just debugging some issue with the RTD builds and noticed you are listed in the maintainers https://readthedocs.org/projects/numba/, I guess this being a result of you kindly instigating this change. Was wondering if you are happy to now hand over the maintenance entirely to the Numba maintainers (i.e. rescind your current Numba RTD maintainer status)? Many thanks :)
> 
> Yes, of course!

Many thanks for confirming!

**Pull request merged**